prow/plugins/config.go

```diff
@@ -798,7 +803,12 @@ func (c *ConfigUpdater) SetDefaults() {
 	}
 
 	for name, spec := range c.Maps {
+		// as a result, namespaces will never be an empty slice (namespace in the slice could be empty string)
+		// and clusters will never be an empty map (map[cluster] could be am empty slice)
 		spec.Namespaces = append([]string{spec.Namespace}, spec.AdditionalNamespaces...)
+		if len(spec.Clusters) == 0 {
```

Comment:
We should validate this case -- it would be surprising if I set `namespace: foo` and then added some _other_ cluster map, only to have my original namespace not honored in the default cluster. If we don't deprecate the original fields, we just need to check that we're not overwriting anything here.